website/build_site.sh

```diff
@@ -95,13 +95,17 @@ for content_filepath in $(find content/ -type f -name '*.md'); do
     # 2. convert relative paths to be relative to the website root instead
     # 3. strip the .md file ending to replace it with a trailing slash
     # (to be consistent with the hugo md -> html conversion)
+    # 4. Strip any unnecessary '/./' path components we introduce by doing that.
 
     # Also keep a backup for comparison/debugging purposes.
 
     sed -i.bak -r \
         -e "s|\]\(([./]*[^:#)]+)#mlos-github-tree-view\)|](https://github.com/microsoft/MLOS/tree/main/${parent_path}/\1)|g" \
-        -e "s|\]\(([./]*[^:)]+)\)|](/MLOS/${parent_path}/\1)|g" \
-        -e "s|\]\(([./]*[^:#)]+)\.md(#[^)]+)?\)|](\1/\2)|g" \
+        -e "s|\]\(([./]*[^:#)]+)(#[^)]*)?\)|](/MLOS/${parent_path}/\1\2)|g" \
+        -e "s|\]\(([./]*[^:#)]+)\.md(#[^)]*)?\)|](\1/\2)|g" \
+        -e "s|\]\(([^)]*)/\./([^)]*)\)|](\1/\2)|g" \
+        -e "s|\]\(([^)]*)/\./([^)]*)\)|](\1/\2)|g" \
+        -e "s|\]\(([^)]*)/\./([^)]*)\)|](\1/\2)|g" \
```

Comment:
replaces `/./` patterns introduced by the previous sed expressions with just `/`.  This isn't strictly necessary, but is nicer.

Reply:
I don't think I understand why this line is here three times.
Generally, as I mentioned in another place, I would prefer to give up on having the markdown links work and not try to maintain a patchwork of regexps to try to make the same file render with two different frameworks.

Reply:
Late night laziness honestly.
Only two are necessary.
One comes from what's originally in the `.md` file links.
One comes from the `dirname` call which resolves to `.` for the top-level items, but just `path/to/subdir` instead of `./path/to/subdir` for others.
The latest changes address that.
Make sense?  Seems okay?